### PR TITLE
Fix permissions for id_rsa

### PR DESCRIPTION
**Pull request by psytale**

### CODE REVIEW

**Comment by tboerger** [comment]
I'm not sure if that is correct. If i remember correctly the folder needs correct permissions as well.

**Comment by msteinert** [comment]
:+1:

I've been using the same workaround in `.drone.yml`...

```
build:
  commands:
    - chmod 600 $HOME/.ssh/id_rsa
```

**Comment by psytale** [comment]
Yeah I wondered if the folder did too hence my branch name, but rethinking pretty sure it doesn't since it doesn't contain any other sensitive files except the private key. This solution certainly works anyway!

**Comment by tboerger** [comment]
My local `.ssh`folder got 600 as well as all private keys.

**Comment by msteinert** [comment]
`600` isn't really a correct permission for a directory. At the very least it should be `700`.

In any case, SSH doesn't care about the directory permissions, just the permissions on the private key. It's actually quite useful for people to be able to access your public key.

**Comment by psytale** [comment]
Have updated for best of both worlds, dir to 700, private key to 600.

**Comment by tboerger** [comment]
Why 700? The machines I've got access to are all set to 600.

**Comment by psytale** [comment]
http://unix.stackexchange.com/questions/21251/how-do-directory-permissions-in-linux-work that was the reference I used after reading due to @msteinert's comment

**Comment by tboerger** [comment]
At least it's the decision of @bradrydzewski what he prefers ;)

File: runner/script/const.go [hunk 1]
```diff
@@ -40,10 +40,11 @@ chmod 0600 $HOME/.netrc
 // private repositories.
 const keyScript = `
 mkdir -p $HOME/.ssh
+chmod 0700 $HOME/.ssh
```
Comment (tboerger) [inline runner/script/const.go]: What about `mkdir -p -m 0700 $HOME/.ssh`?

Comment (psytale) [inline runner/script/const.go]: much better. Updating

**Comment by bradrydzewski** [comment]
Looks good to me.  Let's stick with `0700` since that is what we also use for the `.ssh` directory in our git plugin [1] 

[1] https://github.com/drone-plugins/drone-git/blob/master/main.go#L231

**Comment by bradrydzewski** [comment]
as a side note, I should mention that Drone also uses netrc files. So if you are doing something like a `git clone` or `npm install` or `go get` or whatever to download private repositories during your build, you can use the `https` url to avoid having to manage keys (remove drone key from github repository, re-add it to user account, etc)

this is still a great pull request, but figured I'd advertise this new feature as well :)

**Comment by psytale** [comment]
I'll be honest that's one feature that really made us change to 0.4, and so far after a little bit of teething we are loving it!  Sadly though some dependencies of dependencies insisted on being fetched over ssh!

**Pull request merged**